src/rpc/methods/state.rs

```diff
@@ -2800,3 +2800,84 @@ impl TryFrom<&ChainConfig> for ForkUpgradeParams {
         })
     }
 }
+
+pub enum StateMinerInitialPledgeForSector {}
+
+impl RpcMethod<4> for StateMinerInitialPledgeForSector {
+    const NAME: &'static str = "Filecoin.StateMinerInitialPledgeForSector";
+    const PARAM_NAMES: [&'static str; 4] = [
+        "sector_duration",
+        "sector_size",
+        "verified_size",
+        "tipset_key",
+    ];
+    const API_PATHS: ApiPaths = ApiPaths::V1;
+    const PERMISSION: Permission = Permission::Read;
+
+    type Params = (ChainEpoch, SectorSize, u64, ApiTipsetKey);
+    type Ok = TokenAmount;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (sector_duration, sector_size, verified_size, ApiTipsetKey(tsk)): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        if sector_duration <= 0 {
+            return Err(anyhow::anyhow!("sector duration must be greater than 0").into());
+        }
+        let sec_size: u64 = match sector_size {
+            SectorSize::_2KiB => 2 * 1024,
+            SectorSize::_8MiB => 8 * 1024 * 1024,
+            SectorSize::_512MiB => 512 * 1024 * 1024,
+            SectorSize::_32GiB => 32 * 1024 * 1024 * 1024,
+            SectorSize::_64GiB => 64 * 1024 * 1024 * 1024,
+        };
+
+        if sec_size == 0 {
+            return Err(anyhow::anyhow!("sector size must be non-zero").into());
+        }
+
+        if sec_size < verified_size {
```

Comment:
I would stick to `>` in the test. It reduces cognitive load during Rust/Go code comparison.